tests/harness.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+
+set -Euo pipefail
+trap cleanup SIGINT SIGTERM ERR EXIT
+
+script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+usage() {
```

Comment:
```suggestion
usage() {
  trap - EXIT
```

No need to call cleanup on exit here since it's expected